alanEappen/assignments/cyclicSort/lc287/lc287.js

```diff
@@ -0,0 +1,88 @@
+// Problem Statement #
+
+// We are given an unsorted array containing ‘n+1’ numbers taken from the range 1 to ‘n’. The array has only one duplicate but it can be repeated multiple times. Find that duplicate number without using any extra space. You are, however, allowed to modify the input array.
+
+//given: nums -> unsorted array
+//goal: is to find the duplicate value 
+
+/*
+variables: curPtr = 0
+while loop conditional with curPtr < nums.length
+let swapPtr = nums[curPtr]-1
+The if conditional (nums[curPtr] !== nums[swapPtr])
+[[numscurPtr],[numswapPtr]] = [[numsswapPtr], [[numscurPtr]]]
+else{
+ curPtr++
+}
+
+//make another for loop to find the duplicated value 
+//by going through the sorted nums
+then we make a conditional if nums[i] is equal to he next element
+  if(num[i] == nums[i+1])
+  {
+    return nums[i] 
+  }
+*/
+//edge cases
+/*
+nums.length = 0 return -1
+there is no duplicates return -1
```

Comment:
This approach would be easier to read and understand if it was written out in sentences/phrases instead of code.